### PR TITLE
Resolves #36

### LINKED ISSUE #36: Add year column to necessary resource tables
We need a way to distinguish the hackathon teams, entries, and winners from year to year. Below is an initial todo list of what needs to be implemented to accomplish this feature.

- [x] Create a separate migration that adds a year column of type integer to the team and entry tables. (winners were hardcoded the html 😬 so there's table to update there)
- [x] The migration should also backfill the existing records to have a year set of 2022

Note: I think the data migration piece here can just be done inside an `up` method and then a `down` method can be written that simply removes the column and nothing else.

### PR DESCRIPTION
**Pull request by Kentasmic**
Adds year column to Teams and Entries, backfills old entries to 2022

### CODE REVIEW

File: db/migrate/20230606190606_add_year_to_teams.rb [hunk 1]
```diff
@@ -0,0 +1,15 @@
+class AddYearToTeams < ActiveRecord::Migration[7.0]
+  def up
+    add_column :teams, :year, :integer
+    
+    Team.find_each do |team|
+      puts "Updating Team with id: #{team.id}"
+      team.update! year: 2022
+      puts "Successfully updated Team with id: #{team.id}"
+    end
```
Comment (excid3) [inline db/migrate/20230606190606_add_year_to_teams.rb:9]: You can update every record in the db in a single SQL statement with `Team.update_all(year: 2022)` instead. It'll be much faster than doing each one individually.

**Review by excid3 — CHANGES_REQUESTED** [review]

File: db/migrate/20230606190606_add_year_to_teams.rb (hunk 1 above)
Comment (cjilbert504) [reply db/migrate/20230606190606_add_year_to_teams.rb:9]: Yeah this is true. The one reason I told him to do it like this was to talk about printing the messages about what was updating and the success state. Plus we only had ~80 records so I wasn't concerned about the speed.

Comment (excid3) [reply db/migrate/20230606190606_add_year_to_teams.rb:9]: I'd still put the logging messages around the update_all so you know what's going on.

Also these probably should be separate up and down methods if we're going to modify the existing records like that?

Comment (Kentasmic) [reply db/migrate/20230606190606_add_year_to_teams.rb:9]: Wow, that is fassstttttt.

**Pull request closed without merging**